recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,5 @@
+c_compiler_version:    # [osx]
+  - "17"               # [osx]
+cxx_compiler_version:  # [osx]
+  - "17"               # [osx]
```

Comment:
What's the reason for the pin? Can you add a comment?

Reply:
The same reason we pin clang due to absl linkage errors. I'll add a comment.

Reply:
Do we have any issue where this problem is discussed? absl+clang should work together in all combinations (of course it's possible that something breaks, but then we should raise bugs and get it fixed)

Reply:
Ah, I think I get it. Something about the ABI changed in clang 18, and unless we rebuild abseil with that, we get mismatched ABI (and thus missing symbols). We could just rebuild abseil with clang 18?